core/Pattern.carp

```diff
@@ -1,28 +1,80 @@
 (system-include "carp_pattern.h")
 
 (defmodule Pattern
+
+  (register-type MatchResult "PatternMatchResult" [start Int, end Int])
+  (defmodule MatchResult
+    (defn refmatch2str [ref-matchres] 
```

Comment:
I'd prefer "refmatch-to-str". This "2" naming scheme is not something we use anywhere else in core.

Reply:
More idiomatic names would be `match2str` => `str`, `refmatch2str` => `ref-str`. You don't really want to repeat the name of the type while in the associated module.

Reply:
done